docs/execution/WorkflowExecution.md

```diff
@@ -0,0 +1,96 @@
+
+![](CWP_B_retryable_fail_then_success.png)
+
+However, always using the previous example, if **B** fails from a **non-retryable** failure, and **A** fails from a **retryable** failure, the behavior will once again depend on the failure mode:
+
+If the failure mode is `NoNewCalls`, then A **will not be** retried:
+
+![](NCC_B_fail_A_retryable.png)
+
+If the failure mode is `ContinueWhilePossible`, then A **will be** retried:
+
+![](CWP_B_fail_A_retryable.png)
+
+## Abort
+
+In both Run and Server mode, you have the ability to ask Cromwell to abort a running workflow. This section explains what that entails.
+
+When aborting workflow, either through the [abort endpoint](api/RESTAPI#abort-a-running-workflow) or by terminating the [Cromwell run process](Modes) (if [configured](Configuring#abort) to do so), Cromwell will take the following steps:
+
+- Change the status of the workflow to `Aborting`
+- Do not start any new job going forward
+- Ask every running job to abort
+- Wait for all running jobs to reach completion
+- Finalize the workflow
+- Change the status of the workflow to `Aborted`
+
+The action of aborting a job is backend specific. Cromwell can only ask a backend to abort a job and wait for the backend to notify when it is in fact aborted.
+Note that by the time the backend is asked to abort a job, it might already be successful, or failed, in which case Cromwell will report the job as such.
+
+As an example, the Google backend when asked to abort a job will send an abort request to the Pipelines API. When Pipelines API indicates that the status of the job is aborted, Cromwell will mark it as such.
+However the abort action is entirely dependent on the backend, in this particular case, Pipelines API does not guarantee the success of an abort request (see [Pipelines API documentation on abort](https://cloud.google.com/genomics/reference/rest/v1alpha2/operations/cancel)).
+
+You'll also notice that the workflow is finalized even though being aborted.
+Finalization is the last step in the execution of a workflow and a chance for each backend to do some work before the workflow is terminated.
+Abort won't deny the chance for backends to finalize the workflow.
+
+If a job fails with a retryable failure (e.g is preempted), it will **not** be attempted again when the workflow is aborting.
+
+## Restart
+
+When Cromwell is run in server mode, it is likely that the server will have to be stopped and started again at some point (to upgrade Cromwell to a new version for example).
+In such a case, if the `workflow-restart` configuration option is set to `true` (it is by default), Cromwell will restart all jobs that were in progress.
+
+Specifically it will attempt to reconnect to all running jobs of a workflow.
+The ability to reconnect to an existing job is again backend dependent. The Google Backend and the HPC backends currently support it.
```

Comment:
hmm I wonder if it works in Docker local? 🤔

Reply:
Good question, maybe @kshakir knows ? Can we "recover" a docker job ?

Reply:
I made some changes recently that record the container ID in the execution directory. Since the actually processes should run as children of the Docker daemon, I think this is at least theoretically possible.